zebrad/src/components/mempool.rs

```diff
@@ -142,6 +142,22 @@ impl Service<Request> for Mempool {
         Pin<Box<dyn Future<Output = Result<Self::Response, Self::Error>> + Send + 'static>>;
 
     fn poll_ready(&mut self, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
+        if let Some(tip_action) = self.chain_tip_change.tip_change().now_or_never() {
```

Comment:
As you've already noted in the PR, we'll want to use the new sync method from #2777:
```suggestion
        if let Some(tip_action) = self.chain_tip_change.last_tip_change() {
```

Here's where we're up to with that:
https://github.com/ZcashFoundation/zebra/pull/2777#discussion_r713547511

Reply:
This was done after syncing with main